### PR TITLE
test(ui): unit tests for users tab and user profile icon component

### PR DESCRIPTION
**Pull request by harsh-vador**
### Describe your changes:

I worked on adding unit test cases for below components:
- [x] components/Users/UserProfileIcon
- [x] components/Users/UsersTab

#
### Type of change:

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Users/UserProfileIcon/UserProfileIcon.component.tsx [hunk 1]
```diff
@@ -318,6 +322,7 @@ export const UserProfileIcon = () => {
             <img
               alt="user"
               className="app-bar-user-avatar"
+              data-testid="app-bar-user-avatar"
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/UserProfileIcon/UserProfileIcon.component.tsx:325]:
```suggestion
              data-testid="app-bar-user-profile-pic"
```

File: openmetadata-ui/src/main/resources/ui/src/components/Users/UserProfileIcon/UserProfileIcon.test.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    expect(getByText('label.default')).toBeInTheDocument();
+  });
+
+  it('should display image if image url is valid', () => {
+    const { getByTestId } = render(<UserProfileIcon />);
+
+    expect(getByTestId('app-bar-user-avatar')).toBeInTheDocument();
+  });
+
+  it('should not display avatar if image url is valid', () => {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/UserProfileIcon/UserProfileIcon.test.tsx:86]:
```suggestion
  it('should not display profile pic if image url is invalid', () => {
```

File: openmetadata-ui/src/main/resources/ui/src/components/Users/UserProfileIcon/UserProfileIcon.test.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    (getImageWithResolutionAndFallback as jest.Mock).mockImplementation(
+      () => undefined
+    );
+    const { queryByTestId, getByText } = render(<UserProfileIcon />);
+
+    expect(queryByTestId('app-bar-user-avatar')).not.toBeInTheDocument();
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/UserProfileIcon/UserProfileIcon.test.tsx:92]:
```suggestion
    expect(queryByTestId('app-bar-user-profile-pic')).not.toBeInTheDocument();
```

**Review by Sachin-chaurasiya — COMMENTED** [review]
Left a few comments, thanks

File: openmetadata-ui/src/main/resources/ui/src/components/Users/UserProfileIcon/UserProfileIcon.component.tsx [hunk 4]
```diff
@@ -317,7 +327,8 @@ export const UserProfileIcon = () => {
           {isImgUrlValid ? (
             <img
               alt="user"
-              className="app-bar-user-avatar"
+              className="app-bar-user-profile-pic"
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/UserProfileIcon/UserProfileIcon.component.tsx:330]: As we are changing the class name here so can you update the class name in the styling file as well?

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**